scripts/test/Muon/elemental_analysis/element_button_test.py

```diff
@@ -0,0 +1,143 @@
+        self.assertEqual(self.element.update.call_count, 1)
+
+    @mock.patch('Muon.GUI.ElementalAnalysis.PeriodicTable.periodic_table.QtGui.QBrush')
+    def test_that_brush_is_called_if_element_not_selected_and_bgcolor_not_none(self, mock_brush):
+        self.element.selected = False
+        self.element.bgcolor = 'mybgcolor'
+        self.element.palette = mock.Mock()
+        self.element.setPalette = mock.Mock()
+        self.element.update = mock.Mock()
+        self.element._setBrush()
+
+        self.assertEqual(mock_brush.call_count, 1)
+        mock_brush.assert_called_with('mybgcolor')
+        self.assertEqual(self.element.setPalette.call_count, 1)
+        self.assertEqual(self.element.update.call_count, 1)
+
+    @mock.patch('Muon.GUI.ElementalAnalysis.PeriodicTable.periodic_table.QtGui.QBrush')
+    def test_that_brush_is_called_if_element_not_selected_and_bgcolor_none(self, mock_brush):
+        self.element.selected = False
+        self.element.bgcolor = None
+        self.element.palette = mock.Mock()
+        self.element.setPalette = mock.Mock()
+        self.element.update = mock.Mock()
+        self.element._setBrush()
+
+        self.assertEqual(mock_brush.call_count, 1)
+        mock_brush.assert_called_with()
+        self.assertEqual(self.element.setPalette.call_count, 1)
+        self.assertEqual(self.element.update.call_count, 1)
+
+    @mock.patch('Muon.GUI.ElementalAnalysis.PeriodicTable.periodic_table.QtWidgets.QPushButton.paintEvent')
+    @mock.patch('Muon.GUI.ElementalAnalysis.PeriodicTable.periodic_table.QtGui.QPainter.fillRect')
+    def test_that_paint_event_does_not_call_fillRect_is_brush_is_none(self, mock_fillRect, mock_paintEvent):
+        self.element.brush = None
+        self.element.paintEvent(mock.Mock())
+
+        self.assertEqual(mock_fillRect.call_count, 0)
+
+    @mock.patch('Muon.GUI.ElementalAnalysis.PeriodicTable.periodic_table.QtWidgets.QPushButton.paintEvent')
+    @mock.patch('Muon.GUI.ElementalAnalysis.PeriodicTable.periodic_table.QtGui.QPainter.fillRect')
+    def test_that_paint_event_calls_fillRect_is_brush_is_not_none(self, mock_fillRect, mock_paintEvent):
+        self.element.brush = mock.Mock()
+        self.element.paintEvent(mock.Mock())
+
+        self.assertEqual(mock_fillRect.call_count, 1)
+
+    @mock.patch('Muon.GUI.ElementalAnalysis.PeriodicTable.periodic_table.QtWidgets.QPushButton.paintEvent')
+    @mock.patch('Muon.GUI.ElementalAnalysis.PeriodicTable.periodic_table.QtGui.QPen.setWidth')
+    def test_that_setWidth_called_with_correct_parameters(self, mock_setWidth, mock_paintEvent):
+        self.element.isCurrent = mock.Mock()
+        self.element.isCurrent.return_value = True
+        self.element.paintEvent(mock.Mock())
+        mock_setWidth.assert_called_with(5)
+
+        self.element.isCurrent.return_value = False
+        self.element.paintEvent(mock.Mock())
+        mock_setWidth.assert_called_with(1)
```

Comment:
why 5 and 1?

Reply:
Because the function is called with these parameters in the method `paintEvent`
https://github.com/mantidproject/mantid/blob/39c5e2acb3090ef713ce7efa9c668585a3602993/scripts/Muon/GUI/ElementalAnalysis/PeriodicTable/periodic_table.py#L374

Reply:
can we have comment explaining that